### PR TITLE
Timezone storage of datetime

### PR DESCRIPTION
**Pull request by rtibbles**
## Summary

Adds timezone information to the timestamp stored in the database.

Does so using human readable Timezone names, rather than simply a UTC offset.

Sets Django timezone to the reported OS timezone, and uses current timezone on incoming datetimes (although, this would still be compatible with having a middleware to set this per request).

Adds Django REST framework serializer support for this Field.

## TODO

- [x] Have tests been written for the new code?
- [x] Has documentation been written/updated?
- [X] New dependencies (if any) added to requirements file
- [x] Add an entry to CHANGELOG.rst

### CODE REVIEW

File: requirements/base.txt [hunk 1]
```diff
@@ -20,3 +20,4 @@ jsonfield==2.0.1
 https://github.com/learningequality/morango/archive/e964c9e3d8225fddd5fe086ec3fb0cf124fdd52c.zip#egg=morango
 requests-toolbelt==0.7.1
 clint==0.5.1
+tzlocal==1.4
```
Comment (benjaoming) [inline requirements/base.txt:23]: Oh my, can't believe this is necessary.. but I spend a few minutes on Stackoverflow and realize it really is the only way :|

File: kolibri/logger/serializers.py [hunk 2]
```diff
@@ -101,7 +102,7 @@ def get_currentmasterylog(self, obj):
         except MasteryLog.DoesNotExist:
             return None
 
-class UserSessionLogSerializer(serializers.ModelSerializer):
+class UserSessionLogSerializer(KolibriModelSerializer):
```
Comment (benjaoming) [inline kolibri/logger/serializers.py:105]: Reading the REST framework docs, this really is the only pattern, right? There's no way of specifying a custom field type's serialization on the Field class, right? And no settings? I couldn't find any... just kind of surprised that this is the way to do it :)

File: requirements/base.txt (hunk 1 above)
Comment (rtibbles) [reply requirements/base.txt:23]: I was similarly shocked!

File: kolibri/logger/serializers.py (hunk 2 above)
Comment (rtibbles) [reply kolibri/logger/serializers.py:105]: Yeah, I couldn't find any other way to add a field specific serializer default.

**Comment by rtibbles** [comment]
OK... I tried to do the rename you suggested, but it broke everything (kolibri could no longer run any django commands), so I think I'll just leave it as is for now!

**Comment by benjaoming** [comment]
> OK... I tried to do the rename you suggested, but it broke everything (kolibri could no longer run any django commands)

If you meant `kolibri.core.models.fields`, then it's likely because the `models` package needs to be sensibly crafted.. it's loaded automatically during model introspection. Normally, you'd have to move contents of the old `models.py` to `models/__init__.py`

**Pull request merged**